Python/alerts-to-discord/main.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pprint
+
+# [START v2import]
+from firebase_functions import params
+from firebase_functions.alerts import (
+    app_distribution_fn,
+    crashlytics_fn,
+    performance_fn,
+)
+# [END v2import]
+
+import requests
+
+DISCORD_WEBHOOK_URL = params.SecretParam("DISCORD_WEBHOOK_URL")
+
+
+def post_message_to_discord(bot_name: str,
+                            message_body: str) -> requests.Response:
+    """Posts a message to Discord with Discord's Webhook API.
+
+    Params:
+        bot_name: The bot username to display
+        message_body: The message to post (Discord Markdown)
+    """
+    webhook_url = DISCORD_WEBHOOK_URL.value()
+    if webhook_url == "":
+        raise EnvironmentError(
+            "No webhook URL found. Set the Discord Webhook URL before deploying. "
+            "Learn more about Discord webhooks here: "
+            "https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks"
+        )
+
+    return requests.post(
+        url=webhook_url,
+        json={
+            # Here's what the Discord API supports in the payload:
+            # https://discord.com/developers/docs/resources/webhook#execute-webhook-jsonform-params
+            "username": bot_name,
+            "content": message_body,
+        },
+    )
+
+
+# [START v2Alerts]
+# [START v2CrashlyticsAlertTrigger]
+@crashlytics_fn.on_new_fatal_issue_published(secrets=["DISCORD_WEBHOOK_URL"])
```

Comment:
can do `secrets=DISCORD_WEBHOOK_URL` (referencing the variable, not the string) so that the relationship with `params.SecretParam` above is more clear

Reply:
doesn't work for me https://github.com/firebase/firebase-functions-python/issues/76